src/viking.js

```diff
@@ -1,11 +1,98 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    }
+    return `${this.name} has died in act of combat`;
+  }
+  battleCry() {
+    return `Odin Owns You All!`;
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    }
+    return `A Saxon has died in combat`;
+  }
+}
 
 // War
-class War {}
+class War {
+  vikingArmy = [];
+  saxonArmy = [];
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+  vikingAttack() {
+    // choose random sax
+    let vikingAttacker = this.vikingArmy[
```

Comment:
The naming of the variables is expressive, which is a good practice!
Will the values of these (the current and following) variables be reassigned at some point? If not, would perhaps the use of const be more apt?